lib/sentry.dart

```diff
@@ -56,31 +58,27 @@ class SentryClient {
     Event environmentAttributes,
     bool compressPayload,
     Client httpClient,
-    Clock clock,
+    ClockProvider clock,
```

Comment:
To give people some time to migrate let's make `clock` a `dynamic` parameter. Then in the body of the factory we can do this:

```dart
final ClockProvider clockProvider = clock is ClockProvider
  ? clock
  : clock.get;

// pass clockProvider to SentryClient._
```

This still lets us remove our dependency on `quiver` but won't break existing code. Let's thoroughly document it in the dartdocs though.

Reply:
Done.